packages/@expo/config-plugins/src/utils/plugin-resolver.ts

```diff
@@ -17,7 +17,10 @@ export function resolvePluginForModule(
     // Only resolve `./file.js`, `package/file.js`, `@org/package/file.js`
     const pluginScriptFile = resolveFrom.silent(projectRoot, pluginReference);
     if (pluginScriptFile) {
-      return { isPluginFile: false, filePath: pluginScriptFile };
+      const lastPathComponent = pluginScriptFile.substring(
+        pluginScriptFile.lastIndexOf(path.sep) + 1
+      );
+      return { isPluginFile: lastPathComponent === pluginFileName, filePath: pluginScriptFile };
```

Comment:
This works, but I think we might be able to simplify it even more:

```suggestion
      return {
        isPluginFile: pluginScriptFile.endsWith(path.sep + pluginFileName),
        filePath: pluginScriptFile,
      };
```

Or wouldn't this work for some reason?

Reply:
although maybe not. let's see about the tests